src/levenshtein.cr

```diff
@@ -13,36 +13,68 @@ module Levenshtein
   def self.distance(string1 : String, string2 : String) : Int32
     return 0 if string1 == string2
 
-    s = string1.chars
-    t = string2.chars
-
-    s_size = s.size
-    t_size = t.size
+    s_size = string1.size
+    t_size = string2.size
 
     return t_size if s_size == 0
     return s_size if t_size == 0
 
-    # This is to allocate less memory
-    if t_size > s_size
-      t, s = s, t
-      t_size, s_size = s_size, t_size
-    end
+    if string1.ascii_only? && string2.ascii_only?
```

Comment:
This should check `single_byte_optimizable?` because `ascii_only?` iterates and check that every codepoint is actually an ASCII codepoint.